exonum_client/protobuf_provider/exonum_api.py

```diff
@@ -0,0 +1,61 @@
+"""This module introduces an ExonumApiProvider class, which is capable
+of loading core protobuf files and protobuf files for Rust runtime via
+Exonum node REST API.
+
+This provider is enabled by default.
+"""
+from typing import Optional, Any, List, Dict
+
+from logging import getLogger
+
+from exonum_client.protobuf_loader import ProtoFile, ProtobufProviderInterface
+from exonum_client.api import Api
+
+# pylint: disable=C0103
+logger = getLogger(__name__)
+
+
+class ExonumApiProvider(Api, ProtobufProviderInterface):
+    """ProtobufApi class implements ProtobufProviderInterface interface."""
+
+    def __init__(self, *args: Any, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+
+        self._rust_runtime_url = self.endpoint_prefix + "/runtimes/rust/{}"
+
+    def _get_proto_sources(self, params: Optional[Dict[str, str]] = None) -> List[ProtoFile]:
+        """Retrieves protobuf sources."""
+        proto_sources_endpoint = self._rust_runtime_url.format("proto-sources")
+        response = self.get(proto_sources_endpoint, params=params)
+        if response.status_code != 200 or "application/json" not in response.headers["content-type"]:
+            logger.critical(
+                "Unsuccessfully attempted to retrieve Protobuf sources.\n" "Status code: %s,\n" "body:\n%s",
+                response.status_code,
+                response.content,
+            )
+            raise RuntimeError("Unsuccessfully attempted to retrieve Protobuf sources: {!r}".format(response.content))
```

Comment:
What about to raise a `HTTPError` from the `requests` package here ?

Reply:
We commonly use `RuntimeError` all over the client. Reusing some library-specific error will cause more ambiguity IMHO. Instead, it worth creating client-specific hierarchy of errors derived from `RuntimeError` to be at the same time forward-compatible and more precise.

Reply:
Yeah. I've thought about own errors class as well. But not now 😄